docs/examples/creating-a-crud-api.md

```diff
@@ -0,0 +1,269 @@
+---
+description: |
+  Use HTTP CRUD methods to perform operations on resources. Learn how to use HTTP handlers to create a RESTful API.
+---
+
+The MDN [docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods) are a
+great resource to learn more about HTTP methods. We'll touch on the four
+fundamental methods necessary to create a basic CRUD (create, read, update,
+delete) API here. Additionally, we'll briefly mention CORS requests and how
+`OPTIONS` comes into play.
+
+Using HTTP methods is a common way to create a REST API. Fresh supports common
+HTTP methods in handlers out of the box. Async HTTP requests are also supported.
+Read more about custom handlers [here](/docs/getting-started/custom-handlers).
+
+In this example we'll be creating a small API that uses
+[Deno KV](https://deno.com/kv) to store users in a database.
+
+Our project structure will look like this (in addition to the rest of the Fresh
+code from a new project):
+
+```
+├── routes
+│   └── api
+│       └── users
+│           ├── [id].ts
+│           └── index.ts
+```
+
+In each section about a method, only the relevant handler will be shown. The
+full files are available at the bottom for reference.
+
+## POST
+
+`POST` (create) is used to create a resource.
+
+```tsx
+// /routes/api/index.ts
+async POST(req, _ctx) {
+  const body = await req.text();
+  const user = JSON.parse(body) as User;
+  await upsertUser(user);
+  return new Response(JSON.stringify(user));
+},
+```
+
+Test this with Postman (or your favorite client) with a URL like
+`http://localhost:8000/api/users` and a method of `POST`. Make sure to have a
+payload like:
+
+```json
+{
+  "id": "2",
+  "name": "TestUserName"
+}
+```
+
+You should receive the same thing back:
+
+```json
+{ "id": "2", "name": "TestUserName" }
+```
+
+## GET
+
+`GET` (read) is used to retrieve a resource and is by far the most common HTTP
+method. You can use `GET` to fetch database content, markdown, or static files.
+
+```tsx
+// /routes/api/[id].ts
+async GET(_req, ctx) {
+  const id = ctx.params.id;
+  const user = await getUserById(id);
+  return new Response(JSON.stringify(user));
+},
+```
+
+Let's practice retrieving our user! A `GET` request to
+`http://localhost:8000/api/users/2` should return:
+
+```json
+{ "id": "2", "name": "TestUserName" }
+```
+
+## PUT (and PATCH)
+
+`PUT` (update) and `PATCH` are used to update a resource. While similar, there
+are differences and you should use the one that best suits your use case. Read
+more about HTTP methods on
+[MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods).
```

Comment:
Let's add a sentence here explaining the difference. I had to look it up myself, because I've only worked on GraphQL-based projects in the past few years where only GET + POST is used.

Reply:
Sure, I've added a sentence and will resolve this when I push the new commit.